packages/project-access/src/file/file-search.ts

```diff
@@ -0,0 +1,106 @@
+ */
+declare module 'mem-fs-editor' {
+    type FileMap = { [key: string]: { state: 'modified' | 'deleted' } };
+
+    export interface Editor {
+        /**
+         * Dump files to compare expected result. Provide a cwd for relative path.
+         * See also https://github.com/SBoudrias/mem-fs-editor#dumpcwd-filter for further details.
+         *
+         * @param [cwd] - optional, relative path
+         */
+        dump(cwd?: string): FileMap;
+    }
+}
+
+/**
+ * Creates a filter function that removes files that have been deleted in an instance of a mem-fs-editor.
+ *
+ * @param changes - changes recorded in an instance of a mem-fs-editor
+ * @param filename - relevant filename
+ * @returns a filter function for string arrays
+ */
+function getMemFsFilter(changes: FileMap, filename: string) {
+    const deleted = Object.entries(changes)
+        .filter(([, info]) => info.state === 'deleted')
+        .map(([file]) => join(basename(join(file)) === filename ? dirname(file) : file));
+    return (path: string) => !deleted.find((entry) => path.startsWith(entry));
+}
+
+/**
+ * Concatanates the given list of files with additional files that have been created using a mem-fs-editor and matching the filename.
+ *
+ * @param folders - existing list of folders
+ * @param changes - changes recorded in an instance of a mem-fs-editor
+ * @param filename - relevant filename
+ * @returns Concatanated and deduped list of folders containing the given filename
+ */
+function concatNewFoldersFromMemFs(folders: string[], changes: FileMap, filename: string): string[] {
+    const modified = Object.entries(changes)
+        .filter(([file, info]) => info.state === 'modified' && basename(join(file)) === filename)
+        .map(([file]) => dirname(join(file)));
+    return [...new Set([...folders, ...modified])];
+}
+/**
+ * Search for 'filename' starting from 'root'. Returns array of paths that contain the file.
+ *
+ * @param filename - filename to search
+ * @param root - root folder to start search
+ * @param excludeFolders - list of folder names to exclude (search doesn't traverse into these folders)
+ * @param fs - optional mem-fs-editor instance
+ * @returns - array of paths that contain the filename
+ */
+export function findFiles(filename: string, root: string, excludeFolders: string[], fs?: Editor): Promise<string[]> {
+    return new Promise((resolve, reject) => {
+        const results: string[] = [];
+        const finder = find(root);
+        finder.on('directory', (dir: string, _stat: unknown, stop: () => void) => {
+            const base = basename(dir);
+            if (excludeFolders.includes(base)) {
+                stop();
+            }
+        });
+        finder.on('file', (file: string) => {
+            if (file.endsWith(sep + filename)) {
+                results.push(dirname(file));
+            }
+        });
+        finder.on('end', () => {
+            if (fs) {
+                const changes = fs.dump('');
```

Comment:
@tobiasqueck according to https://github.com/SBoudrias/mem-fs-editor#dumpcwd-filter
the `cwd` argument of `dump()` is optional. Should we remove `''`?

Reply:
No :) Using `''` makes the returned paths absolute, like paths `findit2` returns